resources/puppetlabs/lein-ezbake/template/global/controller.sh.erb

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+
+################
+# Usage: controller.sh <os>
+# The controller script does all of the jar compilation
+# and setup needed to build packages for <os>
+# The os definition is fairly flexible using string matching
+# and currently supports el6, el7, sles11, sles12, debian jessie,
+# debian stretch, ubuntu trusty, ubuntu xenial, and ubuntu yakkety
+################
+build_os=$1
+
+if [ ! -d base ]; then
+        DESTDIR=base bash install.sh install_redhat
+fi
+if [ ! -d systemd_el ]; then
+        cp -r base systemd_el
+        DESTDIR=systemd_el bash install.sh systemd_redhat
+fi
+if [ ! -d old_el ]; then
+        cp -r base old_el
+        DESTDIR=old_el bash install.sh sysv_init_redhat
+fi
+if [ ! -d old_sles ]; then
+        cp -r base old_sles
+        DESTDIR=old_sles bash install.sh sysv_init_suse
+fi
+
+# things are only different if we have docs, deb docs get
+# installed in an unversioned folder but rpm docs get installed
+# in a versioned folder.
+if [ -d ext/docs ]; then
+        if [ ! -d base_deb ]; then
+                DESTDIR=base_deb bash install.sh install_deb
+                if [ ! -d systemd_deb ]; then
+                        cp -r base_deb systemd_deb
+                        DESTDIR=systemd_deb bash install.sh systemd_deb
+                fi
+                if [ ! -d sysvinit_deb ]; then
+                        cp -r base_deb sysvinit_deb
+                        DESTDIR=sysvinit_deb bash install.sh sysv_init_deb
+                fi
+        fi
+else
+        if [ ! -d systemd_deb ]; then
+                cp -r base systemd_deb
+                DESTDIR=systemd_deb bash install.sh systemd_deb
+        fi
+        if [ ! -d sysvinit_deb ]; then
+                cp -r base sysvinit_deb
+                DESTDIR=sysvinit_deb bash install sysv_init_deb
+        fi
+fi
+
+case $build_os in
```

Comment:
It would be great if it were possible to do this programmatically, but I'm not sure how we could make that work. Hmm, we might be able to encode this information in packaging and pass that as environment variables, then at least we'd only need to update it in one place. The directories will still need to be set in here since packaging shouldn't know about those details, and if we update the FPM script to use 2-digit sles versions that would probably help (those I found in a chart online that I'm failing to find again right now), though built-in macros might do something strange if the version isn't how it's expecting it to be